### PR TITLE
Add MlEvent.isEvent() block.

### PR DESCRIPTION
**Pull request by microbit-carlos**
There is a bit more code here than needed, I was cleaning a couple of things as I was going, and adding more debug messages out which I will need for some measurements.

But happy to extra those out to a different PR if we decide to go with a different approach.

### CODE REVIEW

File: pxtextension.ts [hunk 1]
```diff
@@ -35,11 +37,13 @@ class MlEvent {
   //% blockId=mlrunner_on_ml_event
   //% block="on ML event $this"
   onEvent(body: () => void): void {
+    this.hasOnEventHandler = true;
     const wrappedBody = () => {
-      if (!mlrunner.prevAction || mlrunner.prevAction !== this) {
+      const prevAction = mlrunner.Action.prevAction;
+      mlrunner.Action.prevAction = this.eventValue;
+      if (!prevAction || prevAction !== this.eventValue) {
```
Comment (microbit-carlos) [inline pxtextension.ts:44]: With this implementation here we can remove the C++ track of `lastPredictionEventId`?

Comment (microbit-robert) [reply pxtextension.ts:44]: I haven't thought this all the way through, but I think if the user doesn't use the `onEvent` blocks, just the `isEvent` blocks, there's no way the TypeScript layer can know what the last prediction was? The code highlighted would never run in that case.

Comment (carlosperate) [reply pxtextension.ts:44]: Ah, yes, you are right 👍

**Review by microbit-robert — APPROVED** [review]
LGTM. Happy to include the extra debug in this PR.

**Pull request merged**